lib/iris/fileformats/netcdf/loader.py

```diff
@@ -751,6 +763,24 @@ def set(
         finally:
             self.var_dim_chunksizes = old_settings
 
+    @contextmanager
+    def from_file(self) -> None:
+        """
+        Ensures the chunks are loaded in from file variables, else will throw an error.
+
+        Notes
+        -----
+        This function acts as a contextmanager, for use in a 'with' block.
+        """
+        try:
+            old_file_mode = self.file_mode
+            old_var_dim_chunksizes = self.var_dim_chunksizes
+            self.file_mode = True
+            yield
+        finally:
+            self.file_mode = old_file_mode
```

Comment:
We should be doing the mirror of this in `self.set()`, too (sorry I missed that before).